glean-core/csharp/Glean/Utils/Logger.cs

```diff
@@ -0,0 +1,66 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using System;
+using Serilog;
+
+namespace Mozilla.Glean.Utils
+{
+    /// <summary>
+    /// This is a thin wrapper class around the Serilog logger in order to make
+    /// the ergonomics of logging simpler by setting up the formatting and log
+    /// tag.
+    /// </summary>
+    public class Logger
+    {
+        private readonly ILogger log;
+
+        public Logger(string logTag)
+        {
+            var format =
+                string.Format(
+                    "[{{Timestamp:HH:mm:ss}} {{Level:u3}}] {0}:{{Message:lj}}{{NewLine}}{{Exception}}",
+                    logTag);
+
+            // Setup Serilog logging
+            // **Note**: we add `Trace` sink in order to see log messages when
+            // running unit tests more conveniently.
+            log = new LoggerConfiguration()
+                .MinimumLevel.Debug()
```

Comment:
Is this something we could do just in tests, maybe by changing the project file for Tests?
Moreover, how can we enable debugging logging in release builds when we have to investigate problems?

Reply:
This just sets the minimum severity level that will be output by Serilog  but it will still output for all builds.  Think of this more like the Android logging where this means that `Log.d`, `Log.i`, `Log.e` would be output but `Log.v` would not.  So, if we change the minimum level to `Information`, then it would only log messages with severity `Information` or greater.  This is enabled in Release.

Reply:
I would expect this to be `Debug` only in tests, while being configurable at runtime in release (defaulting at Information).

In a release build, where do logs get printed by default? stdout? Are they turned on by default?

Reply:
We don't allow log level configuration for any other binding, why here?  For release or debug builds this will log to the provided sinks, in this case the Console (i.e. stdout), and to Trace (which makes it show up in the application output window in VS).

Reply:
For python, for example, no log gets shown unless we enable them. You also pick the minimum logging level there. The same goes for Rust bindings. Before digging more into this, let me understand it a bit better:

- how can you provide a sink?
- can you turn on/off the logging on a released application?

Reply:
The sinks are provided in the two lines following the one you tagged for this comment.  And no, as this is currently implemented, there isn't a way to turn it on and off from a consuming application.  Just like the Kotlin bindings doesn't turn off the logging to LogCat or provide a way to set a 'sink' for the logging.